### PR TITLE
GEODE-2788: Add official Socket timeout parameter when connecting to servers/locators

### PR DESCRIPTION
**Pull request by masaki-yamakawa**
When connecting from the client to the servers/locators, if the servers/locators is not started, the connection can not be established and a Socket timeout occurs.
This timeout value is 59 seconds by default. This timeout value is too long. This timeout value can be changed by specifying the unofficial parameter "gemfire.PoolImpl.HANDSHAKE_TIMEOUT" in java system property, but I corresponded so that it can be specified by official parameters.
The official parameters are specified by the attribute of Pool.(default:59000 ms)

For example, cache.xml like this:
```xml
<pool name="MyPool" socket-connect-timeout="15000">
  <locator host="localhost" port="10334" />
</pool>
```

API like this:
```java
ClientCacheFactory().addPoolLocator("localhost", 10334).setPoolSocketConnecTimeout(15000);
```
 or
```java
PoolManager.createFactory().addServer("localhost", 40404).setSocketConnectTimeout(15000);
```

### CODE REVIEW

File: geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd [hunk 1]
```diff
@@ -1191,6 +1191,7 @@ As of 6.5 disk-dirs is deprecated on region-attributes. Use disk-store-name inst
         </xsd:complexType>
       </xsd:element>
     </xsd:choice>
+    <xsd:attribute name="socket-connect-timeout" type="xsd:string" use="optional" />
```
Comment (bschuchardt) [inline geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd]: I've opened a discussion on geode-dev about this change.  I would think a new XSD needs to be created.

**Comment by bschuchardt** [comment]
I am running precheckin on this PR and will merge it to develop afterward.

**Comment by bschuchardt** [comment]
Masaki, this change set is breaking unit tests because the new XSD attribute is being put into XML files generated for older versions.  CacheXml80DUnitTest, for instance, is breaking.  l80Please run the precheckin target before posting a fix and ensure that you're not breaking anything.  Thanks

You should also add a test case in CacheXmlGeode10DUnitTest for the new attribute.

org.apache.geode.cache30.CacheXml80DUnitTest > testDefaultConnectionPool FAILED
    org.apache.geode.cache.CacheXmlException: While reading Cache XML file:/export/india2/users/bschuchardt/devel/gfdev/open/geode-core/build/distributedTest/XML_8_0/CacheXml80DUnitTest_testDefaultConnectionPool.xml. Error while parsing XML, caused by org.xml.sax.SAXParseException; lineNumber: 4; columnNumber: 528; Attribute "socket-connect-timeout" must be declared for element type "pool".

        org.xml.sax.SAXParseException; lineNumber: 4; columnNumber: 528; Attribute "socket-connect-timeout" must be declared for element type "pool".

    java.lang.AssertionError: Suspicious strings were written to the log during this run.
    Fix the strings or use IgnoredException.addIgnoredException to ignore.
    -----------------------------------------------------------------------
    Found suspect string in log4j at line 91

    [error 2017/05/10 12:12:11.923 PDT <Test worker> tid=0x13] While reading Cache XML file:/export/india2/users/bschuchardt/devel/gfdev/open/geode-core/build/distributedTest/XML_8_0/CacheXml80DUnitTest_testDefaultConnectionPool.xml. Error while parsing XML

File: geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd (hunk 1 above)
Comment (metatype) [inline geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd]: BTW, if we change the XSD it needs to get posted on geode.apache.org via the asf-site branch of geode-site.

**Comment by masaki-yamakawa** [comment]
@bschuchardt
I am so sorry, I did not understand that the dunit test was not run in 'gradlew test'. I fix it and test with precheckin.

File: geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd (hunk 1 above)
Comment (masaki-yamakawa) [inline geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd]: @metatype
Thank you very much. I understand about posting to geode-site.

**Comment by bschuchardt** [comment]
There is another failure in AnalyzeSerializablesJUnitTest caused by this change.  Certain tests serilalize/deserialize PoolAttributes, so this class is DataSerializable.  This PR changes the serialized form of the class, so the sanctionedSerializables.txt file must be updated to reflect that this change is acceptable:


org.apache.geode.codeAnalysis.AnalyzeSerializablesJUnitTest > testDataSerializables FAILED
    java.lang.AssertionError: Modified classes--------------------------------------------
    org/apache/geode/internal/cache/PoolFactoryImpl$PoolAttributes,2
    fromData,153,2a2bb80034b500052a2bb80034b500072a2bb80034b500092a2bb80034b5000b2a2bb80034b5000c2a2bb80034b5000d2a2bb80034b500112a2bb80035b500102a2bb80035b500142a2bb80034b500182a2bb80034b5001a2a2bb80036b5000a2a2bb80036b500162a2bb80037b5001d2a2bb80038b500212a2bb80038b500222a2bb80034b500152a2bb80036b5001e2a2bb80034b50004b1
    toData,153,2ab400052bb8002f2ab400072bb8002f2ab400092bb8002f2ab4000b2bb8002f2ab4000c2bb8002f2ab4000d2bb8002f2ab400112bb8002f2ab400102bb800302ab400142bb800302ab400182bb8002f2ab4001a2bb8002f2ab4000a2bb800312ab400162bb800312ab4001d2bb800322ab400212bb800332ab400222bb800332ab400152bb8002f2ab4001e2bb800312ab400042bb8002fb1
    :  toData (len=153,expected=145), and fromData (len=153,expected=145)


    If the class is not persisted or sent over the wire add it to the excludedClasses.txt file in the 
    org/apache/geode/codeAnalysis directory.  Otherwise if this doesn't 
    break backward compatibility move the file actualDataSerializables.dat to the codeAnalysis 
    test directory and rename to sanctionedDataSerializables.txt
        at org.junit.Assert.fail(Assert.java:88)
        at org.apache.geode.codeAnalysis.AnalyzeSerializablesJUnitTest.testDataSerializables(AnalyzeSerializablesJUnitTest.java:188)

**Comment by masaki-yamakawa** [comment]
I fixed the test that was failing with precheckin.
As for the modification of XSD, I think that there is no conclusion yet, but at the present time, I am dealing with assumption to change cache-1.0.xsd.

**Comment by bschuchardt** [comment]
Masaki, it looks like there are some conflicts that need to be resolved

**Comment by masaki-yamakawa** [comment]
@bschuchardt
Thank you for pointing it out. I resolved the conflicts.
Travis CI's BUILD failed, but I think it is the effect of GEODE-2959 change.

File: geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd (hunk 1 above)
Comment (masaki-yamakawa) [inline geode-core/src/main/resources/META-INF/schemas/geode.apache.org/schema/cache/cache-1.0.xsd]: @metatype
I changed the XSD of geode-site and posted a pull request.
https://github.com/apache/geode-site/pull/4

**Comment by masaki-yamakawa** [comment]
I fixed LocatorLauncherTest as follows and tested that BUILD succeeded.

```java
  @BeforeClass
  public static void caseSetUp() {
    InternalDistributedSystem ds = InternalDistributedSystem.getConnectedInstance();
    if (ds != null) ds.disconnect();
  }
```

**Comment by masaki-yamakawa** [comment]
I am sorry, once I rebase to confirm the build, travis-ci hangs for some reason.

**Comment by galen-pivotal** [comment]
@masaki-yamakawa  Looks like Travis doesn't think the build has started. I've sent a message to the developer list; someone there should have an idea what the issue is.

**Comment by masaki-yamakawa** [comment]
@galen-pivotal
Thank you very much. I confirmed the message sent to the dev mailing list.

**Comment by galen-pivotal** [comment]
@masaki-yamakawa [Stack overflow](https://stackoverflow.com/questions/17606874/trigger-a-travis-ci-rebuild-without-pushing-a-commit) suggests that you can trigger a build by closing and reopening the PR. I don't have write access, so I can't kick it myself. Sorry for the slow response.

**Comment by masaki-yamakawa** [comment]
@galen-pivotal
Thank you very much. Because there was a conflicting file with the change so far, I merged again, then pushed. Travis-CI worked normally.

**Comment by kohlmu-pivotal** [comment]
@masaki-yamakawa I'm running precheckin. If it passes I'll merge this PR.

**Pull request closed without merging**